contracts/upgradeability/ClassicEternalStorageProxy.sol

```diff
@@ -0,0 +1,35 @@
+pragma solidity 0.4.19;
+
+import "./EternalStorage.sol";
+import "./OwnedUpgradeabilityProxy.sol";
+
+
+contract ClassicEternalStorageProxy is OwnedUpgradeabilityProxy, EternalStorage {
+
+    function () payable public {
+        address _impl = implementation();
+        require(_impl != address(0));
+        bytes4 sig;
+        assembly { sig := calldataload(0) }
+        uint256 len = getSize(sig);
+        assembly {
+            calldatacopy(0x0, 0x0, calldatasize)
```

Comment:
consider to use the way implemented in the original `Proxy`:
```solidity
let ptr := mload(0x40)
calldatacopy(ptr, 0, calldatasize)
let result := delegatecall(gas, _impl, ptr, calldatasize, 0, len)
```

Reply:
Done